metacat-common/src/main/java/com/netflix/metacat/common/QualifiedName.java

```diff
@@ -550,33 +593,88 @@ public String toString() {
         return qualifiedName;
     }
 
+    /**
+     *
+     * Checks if a CharSequence is empty ("") or null.
+     */
+    private static boolean isNullOrEmpty(@Nullable final CharSequence cs) {
+        return cs == null || cs.length() == 0;
+    }
+
     /**
      * Type of the connector resource.
      */
     public enum Type {
         /**
          * Catalog type.
          */
-        CATALOG,
+        CATALOG("CATALOG"),
 
         /**
          * Database type.
          */
-        DATABASE,
+        DATABASE("DATABASE"),
 
         /**
          * Table type.
          */
-        TABLE,
+        TABLE("TABLE"),
 
         /**
          * Partition type.
          */
-        PARTITION,
+        PARTITION("PARTITION"),
 
         /**
          * MView type.
          */
-        MVIEW
+        MVIEW("MVIEW");
+
+
+        /**
+         * TYPE REGEX MAP.
+         */
+        private static final Map<Type, String>
+            REGEX_MAP = new HashMap<Type, String>() {
+            {
+                put(Type.CATALOG,   "^([^\\/]+)$");
+                put(Type.DATABASE,  "^([^\\/]+)\\/([^\\/]+)$");
+                put(Type.TABLE,     "^([^\\/]+)\\/([^\\/]+)\\/([^\\/]+)$");
+                put(Type.MVIEW,     "^([^\\/]+)\\/([^\\/]+)\\/([^\\/]+)\\/([^\\/]+)$");
+            }
+        };
+
+        private final String value;
+
+        /**
+         * Constructor.
+         *
+         * @param value category value.
+         */
+        Type(final String value) {
+            this.value = value;
+        }
+
+
+        public static Map<Type, String> getTypeRegexMap() {
+            return REGEX_MAP;
+        }
+
+        /**
+         * Type create from value.
+         *
+         * @param value string value
+         * @return Type object
+         */
+        public static Type fromValue(final String value) {
```

Comment:
We do not need this if we use valueOf

Reply:
To compare the value with equalsIgnoreCase, also giving more information during the exception.